src/main/java/com/theelixrlabs/healthcare/controller/PatientController.java

```diff
@@ -65,4 +62,16 @@ public ResponseEntity<?> getPatientById(@PathVariable String patientId) throws C
     public ResponseEntity<SuccessResponse<String>> deletePatientById(@PathVariable String patientId) throws CustomException {
         return new ResponseEntity<>(new SuccessResponse<>(true, patientService.deletePatientById(patientId)), HttpStatus.OK);
     }
+
+    /**
+     * Retrieves a list of patients whose first or last name starts with the specified name.
+     *
+     * @param patientName The partial name to search for patients. It matches both first and last names.
+     * @return A ResponseEntity containing a SuccessResponse with the list of matching PatientDTO objects and HTTP status OK.
+     */
+    @GetMapping(EndpointConstants.GET_PATIENTS_BY_NAME_ENDPOINT)
+    public ResponseEntity<SuccessResponse<List<PatientDto>>> getPatientsByName(@RequestParam(PatientConstants.PATIENT_NAME_PARAM) String patientName) {
```

Comment:
can you use patientName in @PathVariable

Reply:
For this task, the requirement is  use @RequestParam. 
The requirement is to get records based on a query parameter (name).